### PR TITLE
Gcpoauth

### PR DESCRIPTION
**Pull request by rosplk**
gcp oauth abuse detection search

### CODE REVIEW

File: stories/gcp_cross_account_activity.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: GCP Cross Account Activity
+id: 0432039c-ef41-4b03-b157-450c25dad1e6
+version: 1
+date: '2020-09-01'
+description: Track when a user assumes an IAM role in another AWS account to obtain
```
Comment (philroyer-phantom) [inline stories/gcp_cross_account_activity.yml:5]: Should this be "GCP" instead of "AWS"?

Comment (rosplk) [reply stories/gcp_cross_account_activity.yml]: This is a story it was fixed to be GCP

File: detections/gcp_detect_oauth_token_abuse.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+date: "2020-09-01"
+description: "This search provides detection of possible GCP Oauth token abuse. GCP Oauth token without time limit can be exfiltrated and reused for keeping access sessions alive without further control of authentication, allowing attackers to access and move laterally."
+how_to_implement: "You must install splunk GCP add-on. This search works with gcp:pubsub:message logs"
+id: a7e9f7bb-8901-4ad0-8d88-0a4ab07b1972
+known_false_positives: "GCP Oauth token abuse detection will only work if there are access policies in place along with audit logs."
+name: "gcp detect oauth token abuse"
+references:
+  - https://www.netskope.com/blog/gcp-oauth-token-hijacking-in-google-cloud-part-1
+  - https://www.netskope.com/blog/gcp-oauth-token-hijacking-in-google-cloud-part-2
+search: '`sourcetype=google:gcp:pubsub:message type.googleapis.com/google.cloud.audit.AuditLog |table protoPayload.@type protoPayload.status.details{}.@type protoPayload.status.details{}.violations{}.callerIp protoPayload.status.details{}.violations{}.type protoPayload.status.message  | `gcp_detect_oauth_token_abuse`'
```
Comment (patel-bhavin) [inline detections/gcp_detect_oauth_token_abuse.yml:11]: the last macro needs "_filter"
the sourcetype in the beginning should be a macro

File: detections/gcp_detect_oauth_token_abuse.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+tags:
+  analytics_story:
+    - "GCP Cross Account Activity"
```
Comment (patel-bhavin) [inline detections/gcp_detect_oauth_token_abuse.yml:14]: updated the formating in the yml

File: stories/gcp_cross_account_activity.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  multiple GCP accounts and roles may be perfectly valid behavior, it may be suspicious
+  when an account requests privileges of an account it has not accessed in the past.
+  After identifying suspicious activities, you can use the provided investigative
+  searches to help you probe more deeply.'
+author: Rod Soto, Splunk
+type: ESCU
+references:
+- https://cloud.google.com/iam/docs/understanding-service-accounts
+tags:
+  analytics_story: GCP Cross Account Activity
```
Comment (patel-bhavin) [inline stories/gcp_cross_account_activity.yml:32]: we have to add this story to atleast one response task for generate.py to run

**Review by patel-bhavin — CHANGES_REQUESTED** [review]

**Review by patel-bhavin — APPROVED** [review]

**Pull request merged**